Source/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSDynamicLinker.c

```diff
@@ -115,16 +116,20 @@ uint32_t bsg_ksdlimageIndexContainingAddress(const uintptr_t address) {
                 if (loadCmd->cmd == LC_SEGMENT) {
                     const struct segment_command *segCmd =
                         (struct segment_command *)cmdPtr;
-                    if (addressWSlide >= segCmd->vmaddr &&
-                        addressWSlide < segCmd->vmaddr + segCmd->vmsize) {
-                        return iImg;
+                    if ( strcmp(segCmd->segname, "__TEXT") == 0 ) {
```

Comment:
Are you sure this is needed? The check for "__TEXT" is happening when the data is populated to the image cache, so I wouldn't have thought it necessary here, so we should reduce the changeset.